accord-core/src/main/java/accord/api/AsyncExecutor.java

```diff
@@ -20,32 +20,102 @@
 
 import java.util.concurrent.Callable;
 import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
 import java.util.function.BiConsumer;
 
+import javax.annotation.Nullable;
+
+import accord.utils.async.AsyncCallbacks;
+import accord.utils.async.AsyncCallbacks.RunOrFail;
 import accord.utils.async.AsyncChain;
-import accord.utils.async.RunnableWithResult;
+import accord.utils.async.AsyncChains;
+import accord.utils.async.Cancellable;
 
 public interface AsyncExecutor extends Executor
 {
-    default AsyncChain<Void> build(Runnable task)
+    // unlike execute, throws no exceptions, nor will not wrap the runnable
```

Comment:
Could you elaborate why _unlike_ `execute`? do you mean unlike `execute(executor, call)` (but then, it does call it internally, so it will have equivalent functionality?